modules/javafx.graphics/src/main/java/javafx/scene/layout/Pane.java

```diff
@@ -104,6 +112,136 @@ public class Pane extends Region {
+                     * including setting origin to null.
+                     *
+                     * Note that users can't do this with the current API's; if a user sets a
+                     * property to a value, the style origin will be USER, even if reset to
+                     * its initial value. So in cases where the property was manually set the
+                     * property will never be removed from the properties map.
+                     */
+
+                    node.getProperties().remove(cssMetaData);
+                }
+            }
+
+            @Override
+            public void invalidated() {
+                if (node.getParent() instanceof Parent p) {
+                    p.requestLayout();
+                }
+            }
+
+            @Override
+            public CssMetaData<Styleable, T> getCssMetaData() {
+                return cssMetaData;
+            }
+
+            @Override
+            public Object getBean() {
+                return node;
+            }
+
+            @Override
+            public String getName() {
+                return propertyName;
+            }
+        };
+    }
+
+    /**
+     * Set the value of a child constraint.
+     *
+     * @param <T> type of the constraint
+     * @param child a child node, cannot be {@code null}
+     * @param cssMetaData a CSS meta data instance, cannot be {@code null}
+     * @param value a value to set, can be {@code null}
+     * @throws NullPointerException when {@code node} or {@cssMetaData} is {@code null}
+     */
+    static <T> void setChildConstraint(Node child, CssMetaData<Styleable, T> cssMetaData, T value) {
+        childConstraintProperty(child, cssMetaData).set(value);
+    }
+
+    /**
+     * Get the value of a child constraint. If the constraint was unset, returns
+     * a default value based on the initial value specified in the CSS meta data.
+     *
+     * @param <T> type of the constraint
+     * @param child a child node, cannot be {@code null}
+     * @param cssMetaData a CSS meta data instance, cannot be {@code null}
+     * @return the current value of the child constraint, can be {@code null} if
+     *     it was unset and the default value was {@code null}
+     * @throws NullPointerException when any argument is {@code null}
+     */
+    static <T> T getChildConstraint(Node child, CssMetaData<Styleable, T> cssMetaData) {
+        Objects.requireNonNull(cssMetaData, "cssMetaData");
+
+        if (child.hasProperties()) {  // implicit null check for child
+            Object value = child.getProperties().get(cssMetaData);
+
+            if (value instanceof StyleableObjectProperty<?> p) {
+                @SuppressWarnings("unchecked")
+                StyleableObjectProperty<T> castProperty = (StyleableObjectProperty<T>) p;
+
+                return castProperty.getValue();
+            }
+
+            if (value != null) {
```

Comment:
How would the value ever be anything other than a `StyleableObjectProperty`, since `setChildConstraint()` always creates a `StyleableObjectProperty`?